crates/kornia-io/src/stream/capture.rs

```diff
@@ -90,20 +91,19 @@ impl StreamCapture {
             .lock()
             .map_err(|_| StreamCaptureError::MutexPoisonError)?;
         if let Some(frame_buffer) = circular_buffer.pop_front() {
-            // TODO: solve the zero copy issue
-            // https://discourse.gstreamer.org/t/zero-copy-video-frames/3856/2
             let buffer = frame_buffer
                 .buffer
                 .map_readable()
                 .map_err(|_| StreamCaptureError::GetBufferError)?;
-            let img = Image::<u8, 3>::new(
-                ImageSize {
-                    width: frame_buffer.width as usize,
-                    height: frame_buffer.height as usize,
-                },
-                buffer.to_owned(),
+
+            let tensor = Tensor::from_shape_slice(
```

Comment:
I don't think it's as simple as this function calls `to_vec` internally. There are two options:
1) Restructure TensorStorage to active a generic for the image that implements `AsRef<u8>` (not sure the consequences os that at all
2) Use `Tensor::from_raw_parts` along with a custom `GstreamerAllocator` (I tried that at some point) which also came with other caveats

let me know if you have any other suggestions